### DISCUSSION TITLE
sudo vcpkg install mosquitto --triplet arm-linux

### DISCUSSION THREAD

**Discussion by ScriptingNotions** [microsoft/vcpkg#33917] [Q&A]
when I try to install mosquitto with "sudo vcpkg install mosquitto --triplet arm-linux"

I get this error:

Computing installation plan...
The following packages will be built and installed:
  * c-ares:arm-linux -> 1.19.1
  * libuv:arm-linux -> 1.46.0
  * libwebsockets:arm-linux -> 4.3.2
    mosquitto:arm-linux -> 2.0.14
  * openssl:arm-linux -> 3.1.2#3
  * pthreads:arm-linux -> 3.0.0#14
  * uthash:arm-linux -> 2.3.0
  * zlib:arm-linux -> 1.3
Additional packages (*) will be modified to complete this operation.
Detecting compiler hash for triplet arm-linux...
error: while detecting compiler information:
The log file content at "/opt/vcpkg/buildtrees/detect_compiler/stdout-arm-linux.log" is:
-- Configuring arm-linux-rel
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:112 (message):
    Command failed: /opt/vcpkg/downloads/tools/cmake-3.27.1-linux/cmake-3.27.1-linux-x86_64/bin/cmake /opt/vcpkg/scripts/detect_compiler -G Ninja -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/opt/vcpkg/packages/detect_compiler_arm-linux -DCMAKE_MAKE_PROGRAM=/opt/vcpkg/downloads/tools/ninja/1.10.2-linux/ninja -DCMAKE_SYSTEM_NAME=Linux -DBUILD_SHARED_LIBS=OFF -DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=/opt/vcpkg/scripts/toolchains/linux.cmake -DVCPKG_TARGET_TRIPLET=arm-linux -DVCPKG_SET_CHARSET_FLAG=ON -DVCPKG_PLATFORM_TOOLSET=external -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON -DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE -DCMAKE_VERBOSE_MAKEFILE=ON -DVCPKG_APPLOCAL_DEPS=OFF -DCMAKE_TOOLCHAIN_FILE=/opt/vcpkg/scripts/buildsystems/vcpkg.cmake -DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON -DVCPKG_CXX_FLAGS= -DVCPKG_CXX_FLAGS_RELEASE= -DVCPKG_CXX_FLAGS_DEBUG= -DVCPKG_C_FLAGS= -DVCPKG_C_FLAGS_RELEASE= -DVCPKG_C_FLAGS_DEBUG= -DVCPKG_CRT_LINKAGE=dynamic -DVCPKG_LINKER_FLAGS= -DVCPKG_LINKER_FLAGS_RELEASE= -DVCPKG_LINKER_FLAGS_DEBUG= -DVCPKG_TARGET_ARCHITECTURE=arm -DCMAKE_INSTALL_LIBDIR:STRING=lib -DCMAKE_INSTALL_BINDIR:STRING=bin -D_VCPKG_ROOT_DIR=/opt/vcpkg -DZ_VCPKG_ROOT_DIR=/opt/vcpkg -D_VCPKG_INSTALLED_DIR=/opt/vcpkg/installed -DVCPKG_MANIFEST_INSTALL=OFF
    Working Directory: /opt/vcpkg/buildtrees/detect_compiler/arm-linux-rel
    Error code: 1
    See logs for more information:
      /opt/vcpkg/buildtrees/detect_compiler/config-arm-linux-rel-CMakeCache.txt.log
      /opt/vcpkg/buildtrees/detect_compiler/config-arm-linux-rel-out.log
      /opt/vcpkg/buildtrees/detect_compiler/config-arm-linux-rel-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_cmake.cmake:344 (vcpkg_execute_required_process)
  scripts/detect_compiler/portfile.cmake:18 (vcpkg_configure_cmake)
  scripts/ports.cmake:147 (include)

error: vcpkg was unable to detect the active compiler's information. See above for the CMake failure output.

Any idea why?

**Comment by dg0yt** [comment]
> * See logs for more information:
>       ...
>       /opt/vcpkg/buildtrees/detect_compiler/config-arm-linux-rel-out.log

which probably has

~~~
-- Cross compiling arm on host x86_64, use cross compiler: arm-linux-gnueabihf-g++/arm-linux-gnueabihf-gcc
-- The C compiler identification is unknown
-- Configuring incomplete, errors occurred!
~~~

which means you are expected to have `arm-linux-gnueabihf-g++` and `arm-linux-gnueabihf-gcc` in your `PATH`.
